azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -992,60 +1015,52 @@ def _send_new_connect_op_down(self):
         def on_connect_complete(op, error):
             this = self_weakref()
             if this:
+                logger.debug(
+                    "{}({}): on_connect_complete error={} state={} never_conencted={} connected={} ".format(
+                        this.name,
+                        op.name,
+                        error,
+                        this.state,
+                        this.never_connected,
+                        this.pipeline_root.connected,
+                    )
+                )
                 if error:
-                    if this.state == ReconnectState.NEVER_CONNECTED:
-                        logger.info(
-                            "{}({}): error on first connection.  Not triggering reconnection".format(
-                                this.name, op.name
-                            )
-                        )
+                    if this.never_connected:
+                        # any error on a first connection is fatal
```

Comment:
>fatal [](start = 61, length = 5)

Is it? It doesn't look fatal

Reply:
"fatal" is maybe a bad error.  I changed it to "transient error" or "permanent error".  I _think_ this is better.

---
In reply to: [439846621](https://github.com/Azure/azure-iot-sdk-python/pull/576#discussion_r439846621) [](ancestors = 439846621)